quadriga/src/main/java/edu/asu/diging/quadriga/web/DisplayCollectionController.java

```diff
@@ -36,43 +45,71 @@ public class DisplayCollectionController {
 
     @Autowired
     private EventGraphService eventGraphService;
+
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+    
+    @Autowired
+    private PredicateManager predicateManager;
     
-   
     
     private Logger logger = LoggerFactory.getLogger(getClass());
 
-    @RequestMapping(value = "/auth/collections/{id}", method = RequestMethod.GET)
-    public String get(@PathVariable String id, Model model) {
-        
+    @RequestMapping(value = "/auth/collections/{collectionId}", method = RequestMethod.GET)
+    public String get(HttpServletRequest request, @PathVariable String collectionId, Model model) {
+
+        // Get collection details
         Collection collection;
         try {
-            collection = collectionManager.findCollection(id);
+            collection = collectionManager.findCollection(collectionId);
             if(collection == null) {
-            	logger.error("Couldn't find collection: ", id);
+            	logger.error("Couldn't find collection: ", collectionId);
                 return "error404Page";
             }
         } catch (InvalidObjectIdException e) {
             logger.error("Couldn't find collection ", e);
             return "error404Page";
         }
-        
-       
-        
+
+        model.addAttribute("collection", collection);
+
+        // Determine page number and size for network pagination
+        int page = 0;
+        int size = 10;
+
+        if (request.getParameter("page") != null && !request.getParameter("page").isEmpty()) {
+            page = Integer.parseInt(request.getParameter("page")) - 1;
```

Comment:
you probably want to catch the runtime exception `parseInt` throws if the value is not actually. a number.